helm/hpcc/templates/_helpers.tpl

```diff
@@ -886,3 +890,179 @@ Pass in dict with root, pod, target and type
+  # Secret names are always required.
+  secretName: {{ .component }}-{{ $exposure }}-{{ $name }}-tls
+  duration: 2160h # 90d
+  renewBefore: 360h # 15d
+  subject:
+    organizations:
+    - HPCC Systems
+  commonName: {{ $name }}.{{ $domain }}
+  isCA: false
+  privateKey:
+    algorithm: RSA
+    encoding: PKCS1
+    size: 2048
+  usages:
+    - server auth
+    - client auth
+  dnsNames:
+ #if servicename parameter is passed we simply create a service entry of that name
+ {{- if .servicename }}
+  - {{ .servicename }}.{{ $domain }}
+ #if service parameter is passed in we are using the component config as a service config entry
+ {{- else if .service -}}
+   {{- $public := and (hasKey .service "public") .service.public -}}
+   {{- if eq $public $externalCert }}
+  - {{ .service.name }}.{{ $domain }}
+   {{- end }}
+ #if services parameter is passed in the component has an array of services to configure
+ {{- else if .services -}}
+  {{- range $service := .services }}
+   {{- $external := and (hasKey $service "external") $service.external -}}
+   {{- if eq $external $externalCert }}
+  - {{ $service.name }}.{{ $domain }}
+   {{- end }}
+  {{- end }}
+ # for internal access the first entry is a wildcard used when the pod is accessed directly (not via a service) for peer to peer communication
+ {{- else if not $externalCert }}
+  - "{{ $name }}.{{ $domain }}"
+ {{- end }}
+  uris:
+  - spiffe://hpcc.{{ $domain }}/{{ .component }}/{{ $name }}
+  # Issuer references are always required.
+  issuerRef:
+    name: {{ $issuer.name }}
+    # We can reference ClusterIssuers by changing the kind here.
+    # The default value is Issuer (i.e. a locally namespaced Issuer)
+    kind: {{ $issuer.kind }}
+    # This is optional since cert-manager will default to this value however
+    # if you are using an external issuer, change this to that issuer group.
+    group: cert-manager.io
+---
+{{- end }}
+{{- end }}
+{{- end }}
+{{- end }}
+
+{{/*
+Experimental: Use certmanager to generate a key for roxie udp encryption.
+A public certificate and private key are generated under /opt/HPCCSystems/secrets/certificates/udp.
+Current udp encryption design would only use the private key.
+Key is in pem format and the private key would need to be extracted.
+*/}}
+{{- define "hpcc.addUDPCertificate" }}
+{{- if .root.Values.certificates -}}
+{{- if .root.Values.certificates.enabled -}}
+{{- $issuer := .root.Values.certificates.issuers.local -}}
+{{- $namespace := .root.Release.Namespace -}}
+{{- $name := .name -}}
+{{- if $issuer }}
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: {{ .component }}-udp-{{ $name }}-cert
+  namespace: {{ $namespace }}
+spec:
+  # Secret names are always required.
+  secretName: {{ .component }}-udp-{{ $name }}-dtls
+  duration: 2160h # 90d
+  renewBefore: 360h # 15d
+  subject:
+    organizations:
+    - HPCC Systems
+  commonName: {{ $name }}.{{ $namespace }}
+  isCA: false
+  privateKey:
+    algorithm: ECDSA
```

Comment:
Yes I just did what I could to get the key size right.  The cert itself isn't intended for roxie to use.  At some point I can play with it or we can come up with a better way of generating those keys.  I just thought this could be a way to do it for now.  When there is a vault available it may have other ways to generate the keys (but would belong in a different PR for sure).

Reply:
So just to clarify, I thought we needed a max key size of 256 and the certificate rules are:
```
	// Size is the key bit size of the corresponding private key for this certificate.
	// If `algorithm` is set to `RSA`, valid values are `2048`, `4096` or `8192`,
	// and will default to `2048` if not specified.
	// If `algorithm` is set to `ECDSA`, valid values are `256`, `384` or `521`,
	// and will default to `256` if not specified.
```
Since only the actual key is being used I can match the algorithm to the desired key size.

Reply:
Also I think Roxie is using AES for UPD encryption.

"AES is a symmetric algorithm which uses the same 128, 192, or 256 bit key for both encryption and decryption (the security of an AES system increases exponentially with key length). "